### ISSUE TITLE
Question: scroll speed by mouse wheel

### ISSUE THREAD

**Issue by ChengCat** [atlas-engineer/nyxt#1100]
Scrolling by mouse wheel works for me, but it's a too slow smooth scroll. How can I make it scroll faster? Thank you!

I am using a recent version from git master: `56a1c161c9b4211ad6664313ad9f6c245e57b0b1`

**Comment by jmercouris** [comment]
You would have to edit `renderer-gtk.lisp` and change the line:

```
(ffi-buffer-enable-smooth-scrolling buffer t)
```

to:

```
(ffi-buffer-enable-smooth-scrolling buffer nil)
```
.

Then, you can recompile, OR you could set a hook to disable it on buffer creation. Ideally we should probably have a buffer slot for this behavior. I too do not favour smooth scrolling.

Thank you for the feedback/question!

**Comment by jmercouris** [comment]
I've updated it in this commit: f10dcaa5

Now you can just do

```
(define-configuration buffer
  ((smooth-scrolling nil)))
```

you can also do it graphically via the usual methods (i.e. describe-slot, describe-class) and clicking `configure`. Hope that helps!

**Comment by ChengCat** [comment]
Thanks! It's now faster, but still not as fast as I'd want. Increasing `scroll-distance` only works for scrolling with keyboard, how can I do the similar to the mouse wheel?

**Comment by jmercouris** [comment]
I don't believe it is possible to increase that, that would be an OS level setting

**Comment by ChengCat** [comment]
I guess it's probably handled by WebKit, and went to track down how it's handled. The wheel event is handled by `ScrollAnimator::handleWheelEvent` (https://trac.webkit.org/browser/webkit/trunk/Source/WebCore/platform/ScrollAnimator.cpp#L168), and that depends on some hard-coded constants here (https://trac.webkit.org/browser/webkit/trunk/Source/WebCore/platform/Scrollbar.h#L51). They should probably expose those constants. I'll maybe try to change the constants for myself when I have more time

Firefox has a setting `mousewheel.default.delta_multiplier_y` in `about:config` for this purpose

**Comment by jmercouris** [comment]
I see, thank you for investigating!

**Comment by JackalSix** [comment]
You can make scroll-distance apply to the mouse wheel by binding the mouse scroll buttons, button4 and button5, to scroll-up and scroll-down respectively. Makes scrolling much snappier if you do that and crank scroll-distance up a bit.

**Comment by hendursaga** [comment]
> They should probably expose those constants.

@ChengCat did you ask/report upstream? did you override them manually, re-compile, and it work for you?

**Comment by jmercouris** [comment]
This is an OS level concern, and the job of the renderer, not much we can do about it.
